### PR TITLE
Add support of get_terminal_size on windows terminal

### PR DESCRIPTION
**Pull request by hhchung**
### Description/Motivation/Screenshots ###
This change make a little modification to make the gef script can be run smoothly on windows terminal (cmder). Due to lack of python module __fcntl__ & __termios__ on windows platform, the function __get_terminal_size__ will meet some errors on windows. Therefore, I add some code to make get_terminal_size can also run on windows terminal.  This modification doesn't change other core functionality, so I think it will not affect the test result. 
By the way, this is awesome project and make me convenient find bugs in program.

#### Screenshot of execution on cmder 
image

### How Has This Been Tested? ###

| Architecture | Yes/No                   | Comments               |
|--------------|:------------------------:|------------------------|
| x86-32       | :heavy_multiplication_x: |  |
| x86-64       | :heavy_multiplication_x: |                        |
| ARM          | :heavy_multiplication_x: |                        | 
| AARCH64      | :heavy_multiplication_x: |  :heavy_check_mark:  |
| MIPS         | :heavy_multiplication_x: |                        |
| POWERPC      | :heavy_multiplication_x: |                        |
| SPARC        | :heavy_multiplication_x: |                        |
| `make tests` | :heavy_multiplication_x: |                        |

### Checklist ###

- [ x ] If my code is a bug fix it targets master, otherwise it targets dev.
- [ x ] My code follows the code style of this project.
- [ x ] My change includes a change to the documentation, if required.
- [ x ] My change adds tests as appropriate.
- [ x ] I have read and agree to the **CONTRIBUTING** document.

### CODE REVIEW

File: gef.py [hunk 1]
```diff
@@ -2867,14 +2865,34 @@ def get_terminal_size():
...
+    if platform.system() == "Windows":
+        from ctypes import windll, create_string_buffer
+        #stdin handle = -10
+        #stdout handle = -11
+        #stderr handle = -12
+        herr = windll.kernel32.GetStdHandle(-12)
+        csbi = create_string_buffer(22)
+        res = windll.kernel32.GetConsoleScreenBufferInfo(herr, csbi)
+        if res:
+            (bufx, bufy, curx, cury, wattr,
```
Comment (Grazfather) [inline gef.py]: just use `_` for the values you don't need

File: gef.py [hunk 2]
```diff
@@ -2867,14 +2865,34 @@ def get_terminal_size():
...
+            return tty_rows, tty_columns
+        else:
+            return 600,100
 
-    try:
-        cmd = struct.unpack("hh", fcntl.ioctl(1, termios.TIOCGWINSZ, "1234"))
-        tty_rows, tty_columns = int(cmd[0]), int(cmd[1])
-        return tty_rows, tty_columns
+    else:
+        import fcntl
```
Comment (Grazfather) [inline gef.py]: Do you get a failure to import these in windows?

Comment (hugsy) [reply gef.py]: IIRC `fcntl` doesn't exist on Windows, yes.

File: gef.py (hunk 1 above)
Comment (hhchung) [reply gef.py]: Hi 
I add `_` for some not actually used variables in the new commit

File: gef.py [hunk 3]
```diff
@@ -2867,14 +2865,34 @@ def get_terminal_size():
     """Return the current terminal size."""
     if is_debug():
         return 600, 100
+    if platform.system() == "Windows":
+        from ctypes import windll, create_string_buffer
+        #stdin handle = -10
+        #stdout handle = -11
+        #stderr handle = -12
+        herr = windll.kernel32.GetStdHandle(-12)
```
Comment (hugsy) [inline gef.py]: Please remove the comments as they don't provide any information about the code itself.
Something like 
```python
hStdErr = -12
herr = windll.kernel32.GetStdHandle(hStdErr)
```
is more explicit.

File: gef.py [hunk 4]
```diff
@@ -2867,14 +2865,34 @@ def get_terminal_size():
...
+        csbi = create_string_buffer(22)
+        res = windll.kernel32.GetConsoleScreenBufferInfo(herr, csbi)
+        if res:
+            (_bufx, _bufy, _curx, _cury, _wattr,
```
Comment (hugsy) [inline gef.py]: The variables `_bufx`, `_bufy`, etc. are not used in the scope, might as well remove them with `_`
```python
_,_,_,_,_, left, top, right, bottom, _, _ = struct.unpack("hhhhHhhhhhh", csbi.raw)
```
is more readable

Comment (hhchung) [reply gef.py]: HI
I added these changes in the new commit.

File: gef.py [hunk 5]
```diff
@@ -2867,14 +2865,32 @@ def get_terminal_size():
     """Return the current terminal size."""
     if is_debug():
         return 600, 100
+    
+    if platform.system() == "Windows":
+        from ctypes import windll, create_string_buffer
+        hStdErr = -12 
+        herr = windll.kernel32.GetStdHandle(hSdErr)
```
Comment (Grazfather) [inline gef.py]: You didn't test your last change: this is missing a 't'

Comment (hhchung) [reply gef.py]: Oh Sorry about that.
I fix this type and test on my windows. I think it should work without error this time
image

**Comment by Grazfather** [comment]
I manually rebased to make sure that the tests pass, and I cleaned it up, but it's merged in dev in commit 957359da12aae86408d21c4c9b2c9b142533ccb9. Thank you for this.

**Pull request closed without merging**